src/OracleService/Protocols/OracleNeoFSProtocol.cs

```diff
@@ -84,28 +84,42 @@ private async Task<string> GetAsync(Uri uri, string host, CancellationToken canc
             var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellation);
             tokenSource.CancelAfter(Settings.Default.NeoFS.Timeout);
             if (ps.Length == 2)
-                return await GetPayloadAsync(client, objectAddr, tokenSource.Token);
-            return await (ps[2] switch
+                return GetPayload(client, objectAddr, tokenSource.Token);
+            return ps[2] switch
             {
-                "range" => GetRangeAsync(client, objectAddr, ps.Skip(3).ToArray(), tokenSource.Token),
-                "header" => GetHeaderAsync(client, objectAddr, tokenSource.Token),
-                "hash" => GetHashAsync(client, objectAddr, ps.Skip(3).ToArray(), tokenSource.Token),
+                "range" => await GetRangeAsync(client, objectAddr, ps.Skip(3).ToArray(), tokenSource.Token),
+                "header" => (OracleResponseCode.Success, await GetHeaderAsync(client, objectAddr, tokenSource.Token)),
```

Comment:
Just for the record: while `GetPayload` and `GetRangeAsync` methods are fixed, the same problem exists with `GetHeader`. Unfortunately, this problem exists on the NeoFS-API level and can't be easily fixed in the `OracleNeoFSProtocol` for now. We'll track this issue in https://github.com/nspcc-dev/neofs-api/issues/171.

Reply:
`GetHashAsync` is also affected, because it also uses `client.GetObjectHeader` under the hood.